core/src/rpc.rs

```diff
@@ -759,15 +813,12 @@ impl JsonRpcRequestProcessor {
                             .runtime
                             .block_on(bigtable_ledger_storage.get_confirmed_block(slot));
                         self.check_bigtable_result(&bigtable_result)?;
-                        return Ok(bigtable_result.ok().map(|confirmed_block| {
-                            confirmed_block.configure(encoding, transaction_details, show_rewards)
-                        }));
+                        return bigtable_result
+                            .map_err(|_| RpcCustomError::BlockNotAvailable { slot }.into());
                     }
                 }
                 self.check_slot_cleaned_up(&result, slot)?;
-                return Ok(result.ok().map(|confirmed_block| {
-                    confirmed_block.configure(encoding, transaction_details, show_rewards)
-                }));
+                return result.map_err(|_| RpcCustomError::BlockNotAvailable { slot }.into());
```

Comment:
These and the following diff cause a behavior change in `getConfirmedBlock`, which I'd strongly prefer to avoid, due to strict partner expectations about this method.
Can we drop this new method, and just call `get_confirmed_block` from `get_inflation_reward`?

Something like:
```
self
    .get_confirmed_block(
        first_confirmed_block_in_epoch,
        Some(RpcEncodingConfigWrapper::Current(Some(RpcConfirmedBlockConfig::rewards_only()))),
    )
```
We could add a helper or a From impl to clean up the ugly Wrapper::Current part

Reply:
Yeah, factoring that out didn't feel quite right, plus I didn't like the name. Will revert